coderd/database/types.go

```diff
@@ -142,3 +143,30 @@ func (a CustomRolePermission) String() string {
 	}
 	return str
 }
+
+// NameOrganizationPair is used as a lookup tuple for custom role rows.
+type NameOrganizationPair struct {
+	Name string `db:"name" json:"name"`
+	// OrganizationID if unset will assume a null column value
+	OrganizationID uuid.UUID `db:"organization_id" json:"organization_id"`
+}
+
+func (*NameOrganizationPair) Scan(_ interface{}) error {
+	return xerrors.Errorf("this should never happen, type 'NameOrganizationPair' should only be used as a parameter")
+}
+
+// Value returns the tuple **literal**
+// To get the literal value to return, you can use the expression syntax in a psql
+// shell.
+//
+//		SELECT ('customrole'::text,'ece79dac-926e-44ca-9790-2ff7c5eb6e0c'::uuid);
+//	To see 'null' option. Using the nil uuid as null to avoid empty string literals for null.
+//		SELECT ('customrole',00000000-0000-0000-0000-000000000000);
+//
+// This value is usually used as an array, NameOrganizationPair[]. You can see
+// what that literal is as well, with proper quoting.
+//
+//	SELECT ARRAY[('customrole'::text,'ece79dac-926e-44ca-9790-2ff7c5eb6e0c'::uuid)];
+func (a NameOrganizationPair) Value() (driver.Value, error) {
+	return fmt.Sprintf(`(%s,%s)`, a.Name, a.OrganizationID.String()), nil
```

Comment:
For the name here probably want to use `https://pkg.go.dev/github.com/lib/pq#QuoteLiteral` unless I'm misunderstanding. This seems like direct sql injection.

Reply:
It cannot be quoted, I was trying that before. This is a parameter argument still, so it will replace `?#`. 

I had considered a sql injection, but can a sql injection happen at a parameter substitution?

Reply:
Ah, good point actually. This is fine, sry

Reply:
I just tried to send in some invalid sql with `--` and `;` as the names, and the query still works.

The `defaultValueConverter` which implments this function for all primitive types just does the literal string value without quotes or escapes: https://github.com/golang/go/blob/master/src/database/sql/driver/types.go#L291-L293

That is the function behavior I am overriding. :+1: